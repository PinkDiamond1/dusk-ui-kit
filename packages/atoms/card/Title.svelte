<div class="{$$props.class || ''} duk-card__title">
  <slot />
</div>
