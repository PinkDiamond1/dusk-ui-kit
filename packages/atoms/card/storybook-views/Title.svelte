<script>
  import Card, { Content, Title } from "../";
  export let variant;
  export let elevation;
  export let content;
  export let title;
  export let size;
</script>

<Card {variant} {elevation} {size} style="width: 305px">
  <Title><h3>{title}</h3></Title>
  <Content>{content}</Content>
</Card>
