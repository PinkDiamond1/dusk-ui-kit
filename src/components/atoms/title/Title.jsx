/* @flow */
import * as React from 'react'

type Props = {
  className: string,
  children: React.Node,
}

const Title = (props: Props): React.Element<*> => {
  const { className, children } = props
  const updatedClassName = `__duk-subtitle ${className}`
  return <h3 className={updatedClassName}>{children}</h3>
}

Title.defaultProps = {
  className: '',
}

export default Title
