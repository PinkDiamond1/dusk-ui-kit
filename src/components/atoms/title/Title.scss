@import "../../../styles/theme/dusk/variables";

.__duk-title {

}
