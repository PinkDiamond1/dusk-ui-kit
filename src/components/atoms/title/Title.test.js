import React from 'react'
import renderer from 'react-test-renderer'

import Title from './Title'

it('test to see if the Title renders correctly', () => {
  const tree = renderer.create(<Title>Smoke Test</Title>).toJSON()
  expect(tree).toMatchSnapshot()
})
