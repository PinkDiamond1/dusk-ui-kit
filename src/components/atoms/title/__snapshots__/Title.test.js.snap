// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Title renders correctly 1`] = `
<h3
  className="__duk-subtitle "
>
  Smoke Test
</h3>
`;
